src/foam/nanos/auth/AuthService.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
+  package: 'foam.nanos.auth',
+  name: 'AuthService',
+	// javaImports: ['foam.core.X', 'javax.security.auth.login.LoginException'],
+  methods: [
+		{
+  		name: 'generateChallenge',
+			javaReturns: 'String',
```

Comment:
Indent with 2 spaces as per the FOAM Coding Standard:
https://github.com/foam-framework/foam/wiki/FOAM2-Style-Guidelines